Port to python3

ayushnawal opened the pull request:
Tested.
All set.

ayushnawal committed:
update sugargame to v1
.3

ayushnawal committed:
add gitignore

ayushnawal committed:
port to python3

ayushnawal committed:
update readme

Aniket21mathur commented:
Thanks. Reviewed. No comments.

ayushnawal commented:
@quozl please review!

quozl commented:
Thanks, but given your other pull requests involving the Sugargame API, I'd prefer to see you fix them before I review any further changes.

ayushnawal commented:
I reviewed and fixed them as you suggested, you can review these changes now

chimosky commented on NumRush.py:
```diff
@@ -38,7 +38,7 @@ def randomOperation(maxi):
     elif (pick == 3):
         x = x * y
-        return 3, x / y, x, y
+        return 3, x // y, x, y
```
Did you confirm if this is supposed to be int division?

ayushnawal replied:
@chimosky Yes, as `x` and `y` both are integers, In Python2, dividing two ints uses integer division

In Python3, dividing two ints results in a float, but using // acts as integer division.

chimosky replied:
Yeah I checked and it was an int hence I resolved the conversation.

srevinsaju reviewed (changes requested):
Thanks @ayushnawal 
Tested, 

* Activity does not terminate properly. This creates an everrunning background process similar to Conzoco activities. On trying to force kill the activiity (ctrl+c) , it creates a `LookupError` regarding dbus.

 found some general bugs _that might not have arisen due to porting_ but, yeah:

* Unable to press / interact with stop button when the countdown goes on. (this might not be an important bug, but of general notice)

quozl commented:
>     * Activity does not terminate properly.

Possibly because the stop button does not have a callback that signals the event loop to terminate.

See the Sugargame test activity for how to do this.

ayushnawal committed:
Adding callback to terminate activity properly

ayushnawal commented:
Terminating for me properly now.

@srevinsaju @quozl please review

srevinsaju commented:
Thanks @ayushnawal, the error's fixed.

quozl commented:
Thanks  Reviewed as at 240e47c.

* [x] variable naming; crashed usually means failure, and a request to stop is not a failure; consider the name as used by the Sugargame test activity,
* [x] although the stop button or accelerator may be used, the three event loops complete a display update and clock delay before returning; it is better to return as soon as the activity is asked to stop,
* [x] `setup.py` is still Python 2,
* [x] `NumRush.py` still begins with a Python version independent hash bang,
* [x] README.md refers to _Numberrush_, but the name of the activity in `activity.info` and elsewhere is different.

Hope that helps.  I didn't have time for a full review.

ayushnawal committed:
Change shebang

replaced crashed with running

ayushnawal committed:
Update README.md

ayushnawal committed:
return as soon as asked to stop

ayushnawal commented:
Fixes done as suggested. Please review @quozl 

> README.md refers to Numberrush, but the name of the activity in activity.info and elsewhere is different.

I looked at `activities.sugarlabs.org` , named it as it was there.

quozl commented:
Thanks.  Reviewed again.

* [ ] the ctrl+q accelerator or stop button does not work during the `newGameAnimation`, because the GTK event source is not being pumped; Sugar pygame activities can't use `pygame.time.wait` for this reason; instead delays have to be implemented by running the event loop until a time has expired, as I mentioned in another pull request.

quozl commented:
Ah, [there it is](https://github.com/sugarlabs/triples/pull/7#issuecomment-578349564).  My comment there shows how to solve the above problem and keep the game user experience the same.

ayushnawal committed:
Fix quit event during newGameAnimation

ayushnawal commented:
Fixed it also :grin: 

Thanks for the help

Tried to improve code quality, removed repetitive code using a function

Tested. all changes are done. 

Please review

quozl commented:
You are yet to
* [ ] run the event loop until a time has expired, as I mentioned in https://github.com/sugarlabs/triples/pull/7#issuecomment-578349564.

As the code stands now, ctrl+q is ignored during each call to pygame.time.wait.

ayushnawal committed:
Fix quit event during newGameAnimation

ayushnawal commented:
Fixed. optimized.

@quozl, please review

quozl commented:
Thanks. Heavy CPU usage during numGameAnimation, please fix.

ayushnawal committed:
Manage frame rate during newGameAnimation

CPU was heavily used during newGameAnimation, fixed by setting fps=30

ayushnawal commented:
Thanks, Fixed it :grin: 

verified using `htop`

@quozl

quozl commented:
Thanks.  Tested.

* [ ] the high score is not kept from one instance to the next; in reviewing this activity on 2018-01-29 it was said "make sure that stopping and resuming the activity returns to the same game state; this is what read_file and write_file are for", but other activities use a file in the instance directory for  keeping a high score,
* [ ] the box can be moved off the right edge by pressing the right arrow key repeatedly,
* [ ] the falling speed of the numbers increases in a step-wise fashion as the game progresses, and the steps seem large,
* [ ] when using the non-activity mode (`python3 NumRush.py`) the window has a window manager border which causes the box to be invisible.

https://github.com/Gr33nMayhem/NumberRush/issues has some other issues yet to be worked on.

JuiP commented:
@ayushnawal ping

srevinsaju commented:
@JuiP , seems like we lost the push on this PR. If you would like to continue this PR, then you may do it.

sourabhaa commented:
@srevinsaju @quozl @chimosky I'm interested to complete this PR. I cloned it in `Activities` directory, but I'm not able to run the activity from the homepage. The activity icon doesn't show up on home page. Am I missing anything?

quozl commented:
That is because as at 7eeaaef ("update sugargame to v1 .3") activity/activity.info exec is not yet ported.

There's also a missing end of line mark in the file, and my shell prompt is misplaced after I used cat.  ;-)

chimosky commented:
> @srevinsaju @quozl @chimosky I'm interested to complete this PR. I cloned it in `Activities` directory, but I'm not able to run the activity from the homepage. The activity icon doesn't show up on home page. Am I missing anything?

Thanks for picking this up, what's left is mentioned in https://github.com/sugarlabs/numberrush-activity/pull/1#issuecomment-600385810 and while looking at the other repo, you can keep the flake8 fixes to a different PR.
